core/src/main/scala/org/ensime/core/debug/MonitorOutput.scala

```diff
@@ -15,20 +15,24 @@ private class MonitorOutput(val inStream: InputStream, broadcaster: ActorRef) ex
 
   @volatile var finished = false
 
-  // TODO This should have a stop method
+  def stopExecution(): Unit = {
+    finished = true
```

Comment:
You need to interrupt the thread as well - otherwise it can be blocked not receiving a byte forever.
I then normally add 'interruptedException to the catch block and silently ignore it.